dist/tools/buildsystem_sanity_check/check.sh

```diff
@@ -41,11 +41,39 @@ check_not_parsing_features() {
     git -C "${RIOTBASE}" grep "${patterns[@]}" -- "${pathspec[@]}"
 }
 
+# Some variables do not need to be exported and even cause issues when being
+# exported because they are evaluated even when not needed.
+#
+# Currently this blacklists exported variables instead of whitelisting or
+# providing a mechanism for handling it.
+# It just keep things not exported anymore in the future.
+UNEXPORTED_VARIABLES=()
+UNEXPORTED_VARIABLES+=('FLASHFILE')
+UNEXPORTED_VARIABLES+=('TERMPROG' 'TERMFLAGS')
+check_not_exporting_variables() {
+    local patterns=()
+    local pathspec=()
+
+    for variable in "${UNEXPORTED_VARIABLES[@]}"; do
+        patterns+=(-e "export ${variable}")
```

Comment:
```suggestion
        patterns+=(-e "export[[:blank:]]\+${variable}")
```

Reply:
I indeed forgot the `\+` and including all whitespaces is safer.
There should not be tabs but does not hurt.

It should be supported by every versions as it is `posix` regex.